tests/vklayertests_gpu.cpp

```diff
@@ -1043,137 +1044,181 @@ TEST_F(VkGpuAssistedLayerTest, GpuBufferDeviceAddressOOB) {
+        }
+    }
+    // descriptor set version
+    {
+        OneOffDescriptorSet descriptor_set(m_device, {{0, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_ALL, nullptr}});
+
+        const VkPipelineLayoutObj pipeline_layout(m_device, {&descriptor_set.layout_});
+        VkDescriptorBufferInfo buffer_test_buffer_info[2] = {};
+        buffer_test_buffer_info[0].buffer = buffer0.handle();
+        buffer_test_buffer_info[0].offset = 0;
+        buffer_test_buffer_info[0].range = sizeof(uint32_t);
+
+        auto descriptor_write = LvlInitStruct<VkWriteDescriptorSet>();
+        descriptor_write.dstSet = descriptor_set.set_;
+        descriptor_write.dstBinding = 0;
+        descriptor_write.descriptorCount = 1;
+        descriptor_write.descriptorType = VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER;
+        descriptor_write.pBufferInfo = buffer_test_buffer_info;
+        vk::UpdateDescriptorSets(m_device->device(), 1, &descriptor_write, 0, NULL);
+
+        char const *shader_source = R"glsl(
+            #version 450
+            #extension GL_EXT_buffer_reference : enable
+            layout(buffer_reference, buffer_reference_align = 16) buffer bufStruct;
+            layout(set = 0, binding = 0) uniform ufoo {
+                bufStruct data;
+                int nWrites;
+            } u_info;
+            layout(buffer_reference, std140) buffer bufStruct {
+                int a[4];
+            };
+            void main() {
+                for (int i=0; i < u_info.nWrites; ++i) {
+                    u_info.data.a[i] = 0xdeadca71;
+                }
+            }
+        )glsl";
+        VkShaderObj vs(this, shader_source, VK_SHADER_STAGE_VERTEX_BIT, SPV_ENV_VULKAN_1_0, SPV_SOURCE_GLSL, nullptr, "main", true);
+
+        VkPipelineObj pipe(m_device);
+        pipe.AddShader(&vs);
+        pipe.AddDefaultColorAttachment();
+        auto err = pipe.CreateVKPipeline(pipeline_layout.handle(), renderPass());
+        ASSERT_VK_SUCCESS(err);
+
+        m_commandBuffer->begin(&begin_info);
+        m_commandBuffer->BeginRenderPass(m_renderPassBeginInfo);
+        vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipe.handle());
+        vk::CmdBindDescriptorSets(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout.handle(), 0, 1,
+                                  &descriptor_set.set_, 0, nullptr);
+        vk::CmdSetViewport(m_commandBuffer->handle(), 0, 1, &viewport);
+        vk::CmdSetScissor(m_commandBuffer->handle(), 0, 1, &scissors);
+        vk::CmdDraw(m_commandBuffer->handle(), 3, 1, 0, 0);
+        vk::CmdEndRenderPass(m_commandBuffer->handle());
+        m_commandBuffer->end();
+
+        for (const auto &test : testcases) {
+            auto *data = static_cast<VkDeviceAddress *>(buffer0.memory().map());
+            data[0] = test.push_constants[0];
+            data[1] = test.push_constants[1];
+            buffer0.memory().unmap();
+
+            if (test.error.size() > 0) {
```

Comment:
Nit
```suggestion
            if (!test.error.empty()) {
```
(Same thing below)